.github/workflows/pre-commit-run.yml

```diff
@@ -0,0 +1,58 @@
+name: Pre-commit Checks
+
+#######
+# Runs the repository's pre-commit hooks.
+#######
+
+on:
+  workflow_call:
+    inputs:
+      python-version:
+        description: 'Python version to use; defaults to latest Python release.'
+        default: '3.X'
+        type: string
+      repository:
+        description: 'GitHub repository to checkout; defaults to repo running this workflow.'
+        default: ${{ github.repository }}
+        type: string
+      pre-commit-source:
+        description: 'The arguments for `pip install` to install pre-commit; use ./path/to/package[dev] for the repo''s pinned version.'
+        default: '.[dev]'
+        type: string
+      runner-os:
+        description: 'Runner OS to use to run tox; defaults to ubuntu-latest'
+        default: 'ubuntu-latest'
+        type: string
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  pre-commit:
+    name: Pre-commit Checks
+    runs-on: ${{ inputs.runner-os }}
+    timeout-minutes: 10
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v3.3.0
+      with:
+        repository: ${{ inputs.repository }}
+
+    - name: Set up Python
+      uses: actions/setup-python@v4.5.0
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Cache pre-commit
+      uses: actions/cache@v3.2.4
+      with:
+        path: ~/.cache/pre-commit
+        key: pre-commit-3|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
+
+    - name: Install pre-commit
+      run: python -m pip install ${{ inputs.pre-commit-source }}
+
+    - name: Run pre-commit
+      run: pre-commit run --show-diff-on-failure --color=always --all-files
```

Comment:
Why a manual install and execution, rather than the action provided by pre-commit itself?

Reply:
Just so we use the version of `pre-commit` that's pinned for the repo and CI doesn't start reporting errors that wouldn't be reported locally due to versions differences.

I'm mostly ambivalent (given the likelihood of issue is quite low), though, if you'd rather continue using the action from `pre-commit`.

Reply:
I guess it's a little weird to be rolling our own when an existing battle-proven action exists... but the fact that we'll be using the exact version pinned in our CI config is one less potential incompatibility between dev and test, so I can live with the duplication.

Reply:
FWIW, I literally just copied `pre-commit`'s current [implementation](https://github.com/pre-commit/action/blob/main/action.yml).